Task2/T2S1.py

```diff
@@ -0,0 +1,79 @@
+import pandas as pd
+
+titles = ['Mr', 'Mrs', 'Miss', 'Ms', 'Master',
+          'Madame', 'Mlle',
+          'Captain', 'Colonel', 'Col', 'Major',
+          'Sir', 'Lady',
+          'Dr',
+          'Rev']
+
+older_male = ['Mr', 'Col', 'Dr', 'Rev', 'Sir']
+
+
+def extract_name_data(elem):
+    col = ['Captain', 'Colonel', 'Col', 'Major']
+    mrs = ['Mrs', 'Madame']
+    ms = ['Ms', 'Miss', 'Mlle']
+
+    elem['Surname'] = elem['Name'].split(',')[0].split('(')[0]
```

Comment:
You should consider using regular expression (https://docs.python.org/3/library/re.html) for extracting the parts of the name from single name column.

Though I'm not completely sure that you can describe all combinations of names with finite set of regular expressions, you should try because it will make the code much simpler, easier to reason about, and also to test.

e.g. Extracting first, second and last name from simple full name:


```
import re
m = re.search('(?P<LastName>\w+), Mr (?P<FirstName>\w+) (?P<SecondName>\w+).*', 'Greenfield, Mr William Bertram')

m.group('LastName')
'Greenfield'

m.group('FirstName')
'William'

m.group('SecondName')
'Bertram'

m.groupdict()
{'FirstName': 'William', 'LastName': 'Greenfield', 'SecondName': 'Bertram'}
```

What you could do is have all these regular expressions in the global list variable, have something like this in your function:

```
NAME_REGEXES = [re.compile('(?P<LastName>\w+), Mr (?P<FirstName>\w+) (?P<SecondName>\w+).*')]
def extract_name_data(element):
    # sub-optimum search for first regular expression that matches this elements name
    match = next(regex.match(element['Name']) for regex in NAME_REGEXES if regex.match(element['Name']))
    # add all named groups to this element if match is found
    if match:
        for key, value in m.groupdict().items():
            element[key] = value

    return element  
```

This way, handling a new name variation is as easy as adding a new regular expression to NAME_REGEXES, and you will avoid all these ugly if else clauses :)